__tests__/actions.test.ts

```diff
@@ -61,24 +61,33 @@ it("runs command actions", async () => {
     connectionString: TEST_DATABASE_URL,
     afterAllMigrations: [{ _: "command", command: "touch testCommandAction" }],
   });
-  const mockedExec: jest.Mock<typeof exec> = exec as any;
-  mockedExec.mockClear();
-  mockedExec.mockImplementationOnce((_cmd, _options, callback) =>
-    callback(null, { stdout: "", stderr: "" }),
-  );
+  const mockedSpawn: jest.Mock<typeof spawn> = spawn as any;
+  mockedSpawn.mockClear();
+  mockedSpawn.mockImplementationOnce((_cmd, _args, _opts): any => {
+    const child = new EventEmitter();
+
+    setImmediate(() => {
+      child.emit("close", 0);
+    });
+
+    return child;
+  });
 
   mockPgClient.query.mockClear();
-  await executeActions(
+  const promise = executeActions(
     parsedSettings,
     false,
     parsedSettings.afterAllMigrations,
   );
+
+  await promise;
+
   expect(mockPgClient.query).toHaveBeenCalledTimes(0);
-  expect(mockedExec).toHaveBeenCalledTimes(1);
-  expect(mockedExec.mock.calls[0][0]).toBe("touch testCommandAction");
-  expect(mockedExec.mock.calls[0][1].env.PATH).toBe(process.env.PATH);
-  expect(mockedExec.mock.calls[0][1].env.GM_SHADOW).toBe(undefined);
-  expect(typeof mockedExec.mock.calls[0][1].env.GM_DBURL).toBe("string");
+  expect(mockedSpawn).toHaveBeenCalledTimes(1);
+  expect(mockedSpawn.mock.calls[0][0]).toBe("touch testCommandAction");
+  expect(mockedSpawn.mock.calls[0][2].env.PATH).toBe(process.env.PATH);
```

Comment:
Is there a suitable assertion we can add on the 2nd arg (in addition to the first and third which we check here)?

Reply:
Presumably that it's an array of length 0.